main.js

```diff
@@ -77,20 +77,20 @@ const log = report => {
 	process.exit(report.errorCount === 0 ? 0 : 1);
 };
 
+const files = (report, predicate) => report.results
+	.filter(predicate)
```

Comment:
`sort` seems more appropriate to me, i.e.:

0. assert `warningCount` and `errorCount` are not `undefined` [*]
1. filter out all results without errors or warnings
2. sort: At first, errors before warnings. Secondly,  _(error|warning)Count_ in descending order.

[*] I do not know, if this is actually needed, maybe `warningCount` and `errorCount` are always numbers.

For example:
```js
.map(result => Object.assign({warningCount: 0, errorCount: 0}, result))
.filter(result => result.errorCount + result.warningCount > 0)
.sort((a, b) => a.errorCount + b.errorCount > 0 ? (b.errorCount - a.errorCount) : (b.warningCount - a.warningCount))
```